Client/Helpers/MsalAuthenticationProvider.cs

```diff
@@ -12,11 +14,11 @@ namespace DeltaQueryClient
     // the GraphSDK team.  It will supports all the types of Client Application as defined by MSAL.
     public class MsalAuthenticationProvider : IAuthenticationProvider
     {
-        private PublicClientApplication _clientApplication;
-        private string[] _scopes;
-        private string _accessToken = null;
```

Comment:
this is very important. **never** store an access token. I could expire (minutes), and then your app would fail. Rather call AcquireTokenSilentAsync which will refresh it (silently) if needed and if possible.